src/react-components/ui-root.js

```diff
@@ -1730,15 +1726,18 @@ class UIRoot extends Component {
               history={this.props.history}
               render={() =>
                 this.renderDialog(RoomSettingsDialog, {
-                  showRoomAccessSettings: this.props.hubChannel.can("update_hub_promotion"),
+                  showPublicRoomSetting: this.props.hubChannel.can("update_hub_promotion"),
                   initialSettings: {
                     name: this.props.hub.name,
                     description: this.props.hub.description,
                     member_permissions: this.props.hub.member_permissions,
                     room_size: this.props.hub.room_size,
-                    allow_promotion: this.props.hub.allow_promotion
+                    allow_promotion: this.props.hub.allow_promotion,
+                    entry_mode: this.props.hub.entry_mode
                   },
-                  onChange: settings => this.props.hubChannel.updateHub(settings)
+                  onChange: settings => this.props.hubChannel.updateHub(settings),
+                  onFetchInvite: () => this.props.hubChannel.fetchInvite(),
```

Comment:
I might just pass in the hubChannel instead and then just call these directly. As implemented when I was reading the component code I was a bit confused since I thought these were callbacks that would trigger when these things happened.

Reply:
I don't like the way we name "callbacks" in react either. We've made it a convention to do things like `onChange` as well. Are you suggestion we reverse that convention?